src/internal/clusterstate/v2.11.0/pjs.go

```diff
@@ -0,0 +1,55 @@
+package v2_11_0
+
+import (
+	"context"
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/migrations"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pctx"
+)
+
+func createPJSSchema(ctx context.Context, env migrations.Env) error {
+	ctx = pctx.Child(ctx, "createPJSSchema")
+	tx := env.Tx
+	_, err := tx.ExecContext(ctx, `
+		CREATE SCHEMA pjs; 
+		CREATE TYPE pjs.job_error_code AS ENUM (
+			'failed', 
+			'disconnected',
+			'canceled'
+		);
+		CREATE TABLE pjs.jobs (
+				id BIGSERIAL PRIMARY KEY,
+				parent BIGINT REFERENCES pjs.jobs(id),
+				
+				program BYTEA NOT NULL,
+				program_hash BYTEA NOT NULL, -- the hash will be used to model the queues
+				
+				error pjs.job_error_code,
+				
+				queued timestamptz DEFAULT CURRENT_TIMESTAMP NOT NULL,
+				processing timestamptz,
+				done timestamptz		
+		);
+		CREATE TYPE pjs.fileset_types AS ENUM (
+			'input', 
+			'output'
+		);
+		-- A job's input and output may be more than one fileset.
+		-- An extra table makes looking up, inserting, and reordering those filesets more efficient.
+		CREATE TABLE pjs.job_filesets (
+			job_id BIGINT REFERENCES pjs.jobs(id),
+			type pjs.fileset_types NOT NULL,
```

Comment:
Is naming the column type going to be a headache?  It seems like it could be special/reserved word?  If it works keep it, if it causes problems during the rest of the implementation let's change it before we ship.